### PR TITLE
Prefs [TASK][26308] Create Plan & report UI

### PR DESCRIPTION
**Pull request by toandv98**
## Related Tickets

- [#Task 26308](https://edu-redmine.sun-asterisk.vn/issues/26308)

## WHAT

- ### Create Plan & report UI

## HOW

- Create layout for plan & report screen
- RecyclerView show all plan & report
- SearchView filter recyclerview item
- FloatingActionButton show message screen

## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Reviewer1 (ToanDV) | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the source code have end-line errors?                   |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| Conventions   | No hard coding, use constants/configuration values           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Conventions   | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

## Notes (Optional)

*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/drawable/bg_circle.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="oval">
+    <solid android:color="@android:color/white" />
+    <size
+        android:width="@dimen/dp_56"
+        android:height="@dimen/dp_56" />
```
Comment (tuanpa-0908) [inline app/src/main/res/drawable/bg_circle.xml]: Thật ra oval tròn không cần size ô ơi

File: app/src/main/res/layout/fragment_report.xml [hunk 2]
```diff
@@ -1,6 +1,52 @@
 <?xml version="1.0" encoding="utf-8"?>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
     android:layout_height="match_parent">
 
+    <include
+        android:id="@+id/app_bar_layout"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/fragment_report.xml]:
```suggestion
        android:id="@+id/layoutAppBar"
```

File: app/src/main/res/layout/item_daily_report.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        app:layout_constraintBottom_toTopOf="@+id/textReport"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textDate" />
+
+    <TextView
+        android:id="@+id/textReport"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/item_daily_report.xml]:
```suggestion
        android:layout_marginStart="@dimen/dp_8"
```

File: app/src/main/res/layout/item_daily_report.xml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textMessage" />
+
+    <TextView
+        android:id="@+id/textSendTime"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="8dp"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/item_daily_report.xml]: Như trên

File: app/src/main/java/com/edu/sun/oereminder/ui/timetracking/TimeTrackingFragment.kt [hunk 5]
```diff
@@ -65,38 +60,39 @@ class TimeTrackingFragment : BaseFragment<View, TimeTrackingPresenter>(), View {
...
-            }
+            presenter.onCheckResult(bundle.getInt(REQUEST_CHECK_IN))
         }
     }
 
-    override fun showOrHideFab(isShow: Boolean) {
-        fabCheckIn.visibility = if (isShow) VISIBLE else GONE
+    override fun updateFab(isShow: Boolean, isCheckIn: Boolean) {
+        fabCheckIn.run {
+            visibility = if (isShow) VISIBLE else GONE
```
Comment (huytq-2422) [inline app/src/main/java/com/edu/sun/oereminder/ui/timetracking/TimeTrackingFragment.kt]:
```suggestion
            isVisible = isShow
```

**Review by tuanpa-0908 — APPROVED** [review]

**Pull request merged**